From fe8442c0461a59f529281d6b63775260ac7487ea Mon Sep 17 00:00:00 2001
From: Robert Yang <liezhi.yang@windriver.com>
Date: Sun, 18 May 2014 18:02:20 +0800
Subject: [PATCH] storage/__init__.py: check /boot/efi before go forward

The EFI System Partition must be mounted on /boot/efi, we need check
this before go forward.

Upstream-Status: Pending

Signed-off-by: Robert Yang <liezhi.yang@windriver.com>
---
 pyanaconda/storage/__init__.py | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/pyanaconda/storage/__init__.py b/pyanaconda/storage/__init__.py
index 5fd747f..3b9874a 100644
--- a/pyanaconda/storage/__init__.py
+++ b/pyanaconda/storage/__init__.py
@@ -1226,6 +1226,14 @@ class Storage(object):
             if mountpoint in mustbeonlinuxfs and (not dev.format.mountable or not dev.format.linuxNative):
                 errors.append(_("The mount point %s must be on a linux file system.") % mountpoint)
 
+        from pyanaconda import platform
+        p = platform.getPlatform(None)
+        if isinstance(p, platform.EFI):
+            if  "/boot/efi" not in filesystems:
+                errors.append(_("The EFI System Partition must be mounted on /boot/efi"))
+            elif filesystems["/boot/efi"].format.type != "efi":
+                errors.append(_("The EFI System Partition must be mounted on /boot/efi"))
+
         return (errors, warnings)
 
     def isProtected(self, device):
-- 
1.8.3.1

